### PR TITLE
dialects: (riscv_func) add custom syntax to mirror func

### PR DESCRIPTION
**Pull request by superlopuh**
Will make some filecheck files with upcoming changes easier to read

### CODE REVIEW

**Review by AntonLydike — COMMENTED** [review]
Can we abstract the func, return and call custom syntax to a python mixin? That would be super nice to have so we can get the same for llvm syntax, and other func-like ops

**Comment by superlopuh** [comment]
We definitely could, I'm not sure that we necessarily want to do that right now. What do you think about me filing that as an issue for later?

**Review by JosseVanDelm — COMMENTED** [review]
Really cool! However, I get a bit the feeling that there's quite a bit of code duplication involved for parsing and printing. Is none of this somewhat shareable with func dialect?

**Comment by superlopuh** [comment]
It seems that @AntonLydike and @JosseVanDelm agree, I'm just not sure that this is the PR to do this factoring in? If one more person tells me to do it I'll do it :)

**Review by adutilleul — APPROVED** [review]
I also have the same feeling regarding the duplication. Since we're importing more and more from standards dialects into the riscv codebase, maintaining duplicated code on the long term is kinda annoying.
However if you feel that we should move on and open an issue for that, that also seems great to me.

**Comment by superlopuh** [comment]
all done, factored code into xdsl.dialects.utils

**Review by nazavode — APPROVED** [review]

File: xdsl/dialects/utils.py
Comment (superlopuh) [inline xdsl/dialects/utils.py:1]: pretty much just moved from func.py

**Review by AntonLydike — APPROVED** [review]
:heart_eyes:

**Pull request merged**